### PR TITLE
Safeway Scraper 

### PR DESCRIPTION
**Pull request by AnudeepYakkala**
Implemented a Python scraper that obtains item and inventory information for Safeway stores. Running with "python3 safeway_scraper.py -test" can be used for debug purposes since it doesn't write to spanner.

### CODE REVIEW

File: capstone/backend/scraper/SafewayScraper.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  items = []
+  for item in soup.findAll('product-item'):
+    try: 
+      ppuString = item.find('span', {'class': 'product-price-qty'}).text.replace(',', '')
+      current = {
+        NAME_KEY: item.find('a', {'class': 'product-title'}).text,
+        PRICE_KEY: float(item.find("span", {'class': 'product-price'}).text[11:]),
+        PPU_KEY: float(ppuString[ppuString.index('$') + 1 : ppuString.index('/') - 1]),
+        UNIT_KEY: ppuString[ppuString.index('/') + 2 : ppuString.index(')')],
+        AVALIBILITY_KEY: 'AVALIABLE'
```
Comment (officialcwang) [inline capstone/backend/scraper/SafewayScraper.py:73]: nit: typo (see earlier comment)

File: capstone/backend/scraper/SafewayScraper.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
+"""
```
Comment (officialcwang) [inline capstone/backend/scraper/SafewayScraper.py:1]: Write tests? (see scraper_test for example)

File: capstone/backend/scraper/scrapers/safeway_scraper.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+def get_store_address(soup):
+  """
+  Returns address of the store given the page source.
+  """
+  address = soup.find('span', {'class': 'reserve-nav__current-instore-text'}).text
+  return address
+
+
+def write_item_info_to_items_table(item_id, item_name, item_brand, item_subtype, item_type):
```
Comment (chen-dawn) [inline capstone/backend/scraper/scrapers/safeway_scraper.py:95]: Can we create a separate write to spanner file with these functions?

File: capstone/backend/scraper/scrapers/safeway_scraper.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  with database.batch() as batch:
+    batch.insert_or_update(
+      table='Stores',
+      columns=store_cols,
+      values=[[store_id, address, store_name]]
+    )
+
+if __name__ == '__main__':
+  # Hard-coded item types.
+  item_types = ['milk', 'paper towels', 'water', 'cookies', 'pencil',
```
[the remark this answers is not in the document]
Comment (AnudeepYakkala) [reply capstone/backend/scraper/scrapers/safeway_scraper.py:130]: Yeah, it is. Should it be different?

[the remark this answers is not in the document]
Comment (chen-dawn) [reply capstone/backend/scraper/scrapers/safeway_scraper.py:130]: nope, this is good.

File: capstone/backend/scraper/scrapers/safeway_scraper.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        write_item_info_to_items_table(item[ID_KEY], item[NAME_KEY], '', '', item_type)
+        write_item_info_to_inventory_table(item[ID_KEY], item[AVALIBILITY_KEY], item[PRICE_KEY], \
+          item[PPU_KEY], item[UNIT_KEY], store_id)
+      f = open('csv/safeway-' + zipcode + '.csv', 'a')
+      writer = csv.DictWriter(f, item.keys())
+      writer.writerow(item)
+      f.close()
+
+if __name__ == "__main__":
+  if (len(sys.argv) > 1) and (str(sys.argv[1]) == 'test'):
```
Comment (chen-dawn) [inline capstone/backend/scraper/scrapers/safeway_scraper.py:121]: use flags: https://abseil.io/docs/python/guides/flags

**Pull request merged**